src/asInput/index.jsx

```diff
@@ -67,6 +74,23 @@ const asInput = (WrappedComponent, inputType = undefined, labelFirst = true) =>
           value: nextProps.value,
         });
       }
+      if (nextProps.isValid !== this.props.isValid && !nextProps.validator) {
+        this.setState({
+          isValid: nextProps.isValid,
+        });
+      }
+      if (nextProps.validationMessage !== this.props.validationMessage && !nextProps.validator) {
+        this.setState({
+          validationMessage: nextProps.validationMessage,
+        });
+      }
+      // If validator goes away, revert to props
```

Comment:
Do we need to handle the case where a new validator is defined in `nextProps`?

Reply:
@jaebradley naw. We don't do any revalidation today when a new validator is set. So validator functions currently expect to only update state when they are called onBlur.